### ISSUE TITLE
Favoritenliste, ggf. auch über Routeneingabe (ICAO)

### ISSUE THREAD

**Issue by Michael-Wellner** [mipastgt/AIPBrowserDE-Public#6]
Wäre es ggf. möglich, einzelne Karten (oder ggf. auch ganze Plätze) zu markieren und in einer Favoritenliste zu speichern, die dann schnell mit einem Klick am linken Rand erreichbar wäre)?
Die Krönung wäre es, wenn man diese Liste z.B. einfach durch Angabe von ICAO-Codes erstellen könnte und ihr vielleicht noch einen Namen geben kann (damit ggf. auch mehrere Listen zu verwalten wären).
In der Praxis könnte man dann quasi eine Liste von Karten in der benötigten Reihenfolge der geflogenen Route erstellen, auf die man während des Fluges dann schnell zugreifen, oder diese im Idealfall sogar mittels der Pfeile oben durchblättern kann.
Beispiel:
Wooge-Borkum:EDWG-EDWL-EDWY-EDWR

**Comment by mipastgt** [comment]
Der zweite Teil dieses Wunsches sollte sich eigentlich mit der bereits [hier](https://github.com/mipastgt/AIPBrowserDE-Public/issues/2#issuecomment-1792404141) beschriebenen Funktion realisieren lassen denn die tut eigentlich genau das.

Die Favoritenliste sehe ich eigentlich unabhängig von dieser Funktion zur Erstellung von PDF-Exporten für ein bestimmtes Flugvorhaben.

**Comment by Michael-Wellner** [comment]
Ja, die Favoritenliste sollte ja interaktiv INNERHALB des Programms genutzt werden, damit eben nicht die externen PDFs verwendet werden müssen.
Gleiches gilt quasi für die "Sonderform" der Favoritenliste als "Routen/Kartenliste", siehe auch die Mail, die ich Ihnen gerade zur Verdeutlichung gesendet habe:

Zu der "Kartenliste"... Im Prinzip wäre das eine spezielle Variante der Favoritenliste, die sich möglichst leicht erzeugen lassen würde.
Wenn wir unsere Flugplanung machen, suchen wir uns ja die entsprechenden Karten gemäß der zu fliegenden Route heraus und studieren diese dann entsprechend. Als "Simulaten" sind wir dann ja nicht im Cockpit sondern am PC unterwegs und haben die Karten (in diesem Fall in Form Ihres Programms) dann zumeist auf einem zweiten Bildschirm geöffnet. Dabei ist es dann hilfreich, wenn diese bereits in der richtigen "Reihenfolge" angeordnet sind, da mal im Flug selber eben nicht unbedingt die Zeit hat innerhalb der Navigation groß zu blättern. Derzeit nutze ich dafür schlicht den Acrobat-Reader mit mehreren Tabs, bei dem ich die einzelnen Karten bereits vorab alle in der notwendigen Reihenfolge geöffnet habe. So brauche ich dann im Flug lediglich immer auf den nächsten Tab zu wechseln, um die nächste Karte aufzurufen.
 
Wie könnte diese Möglichkeit nun in einem Programm realisiert werden?

Ich gebe (z.B. einfach über die bereits vorhandene Suchleiste) eine Route anhande der ICAO-Codes der Flugplätze ein: EDWG-EDWL-EDWR.
Nun erzeugt das Programm selbstständig eine "Favoritenliste" mit den entsprechenden Plätzen als "Baumstruktur" in der Reihenfolge der Plätze quasi in einem eigenen "Ordner" am Navigationsrand:
 
- **Aeronautical Information Publication VFR**
   - ....
- **EDWG-EDWL-EDWR**
   - Wangerooge EDWG
      - EDWG Wangerooge 1
      - EDWG Wangerooge 2
      - AD 2-106
   - Langeoog EDWL
      - EDWL Langeoog 1
      - EDWL Langeoog 2
      - AD 2-60
   - Borkum EDWR
      - EDWR Borkum 1
      - EDWR Borkum 2
      - AD 2-19

Wählt man nun eine Karte (in der Regel ja die erste der Liste aus und blättert in der Anzeige oben weiter, so erhält man nun entsprechend der Reihenfolge in der Liste die entsprechenden Karten, die man innerhalb des Fluges jeweils benötigt.
Ein zusätzliches "Schmankerl" wäre natürlich das man beim ersten Platz (Abflug) in der Reihenfolge zuerst Karte 2 (Rollkarte) und dann Karte 1 (Anflugkarte) hätte - sich also die Reihenfolge der Karten dort einfach "umdreht". Am letzten Platz (Anflug) und an denen innerhalb der Route würde die Reihenfolge dann ganz normal bleiben.

**Comment by Michael-Wellner** [comment]
Man könnte diese Funktionen (Favoriten/Routen) für eine bessere Übersichtlichkeit natürlich auch zusätzlich noch in einen jeweiligen übergeordneten Ordner legen:


    Aeronautical Information Publication VFR
        ....
    Favoriten
        Berlin
          ...
        Bremen
          ...
    Routen
        EDWG-EDWL-EDWR
            Wangerooge EDWG
                AD 2-106
                EDWG Wangerooge 2
                EDWG Wangerooge 1
            Langeoog EDWL
                EDWL Langeoog 1
                EDWL Langeoog 2
                AD 2-60
            Borkum EDWR
                EDWR Borkum 1
                EDWR Borkum 2
                AD 2-19
